boxsdk/object/file.py

```diff
@@ -72,12 +72,26 @@ def update_contents_with_stream(self, file_stream, etag=None):
             If specified, instruct the Box API to update the item only if the current version's etag matches.
         :type etag:
             `unicode` or None
+        :param preflight_check:
+            If specified, preflight check will be performed before actually uploading the file.
+        :type preflight_check:
+            `bool`
+        :param size:
```

Comment:
I think it's unintuitive for there to be a separate parameter called size which is only relevant for preflight, yet the parameter name doesn't mention it. I would suggest either renaming size to expected_size_for_preflight or, alternatively, changing preflight_check's type to a dictionary that can optionally contain a size key. Then if preflight_check is None, skip the check. If it's not None, do the check, and see if there's a size key.